### ISSUE TITLE
[java] CloseResource: false positive with method call

### ISSUE THREAD

**Issue by Han-Cui** [pmd/pmd#4623]
**Affects PMD Version:** 7.0.0

**Rule:** CloseResource

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#closeresource>

**Description:**
PMD warns in line 11, while the resource _c_ gets closed in the function _closeConnection_.
**Code Sample demonstrating the issue:**
Maybe a false positive:

```java
import java.sql.Connection;
import net.sourceforge.pmd.lang.java.rule.errorprone.closeresource.Pool;

public class Foo {

    void closeConnection(Connection c) {
        c.close();
    }

    void bar(Pool pool) {
        Connection c = pool.getConnection();
        try {
        } catch (Exception e) {
        } finally {
            closeConnection(c);
        }
    }

}
```

Compliant code:

```java
import java.sql.Connection;
import net.sourceforge.pmd.lang.java.rule.errorprone.closeresource.Pool;

public class Foo {
    void bar(Pool pool) {
        Connection c = pool.getConnection();
        try {
        } catch (Exception e) {
        } finally {
            c.close();
        }
    }
}
```

**Expected outcome:**
No warnings.

**Running PMD through:** *[CLI]*

**Comment by adangel** [comment]
You could configure the rule property "closeTargets" like this:

```xml
<rule ref="category/java/errorprone.xml/CloseResource">
    <properties>
        <property name="closeTargets" value="closeConnection" />
    </properties>
</rule>
```

However, since this is all within one class, it would be nice if PMD could detect this on its own (through some basic call flow).
